check_source.py

```diff
@@ -166,28 +166,7 @@ def check_source_submission(self, source_project, source_package, source_revisio
                     inair_renamed = target_package != match.group(1)
 
         if not self.source_has_correct_maintainers(source_project):
-            declined_msg = (
-                'This request cannot be accepted unless %s is a maintainer of %s.' %
-                (self.required_maintainer, source_project)
-            )
-
-            try:
-                add_roles = get_request_list(self.apiurl, source_project,
-                    req_state=['new', 'review'], req_type='add_role')
-                add_roles = list(filter(self.select_add_role, add_roles))
-                if len(add_roles) > 0:
-                    add_role_reqid = add_roles[0].reqid
-                else:
-                    add_role_msg = 'Created automatically from request %s' % self.request.reqid
-                    add_role_reqid = create_add_role_request(self.apiurl, source_project, self.required_maintainer,
-                                                            'maintainer', message=add_role_msg)
-                declined_msg += ' Created the add_role request %s for addressing this problem.' % add_role_reqid
-            except HTTPError as e:
-                self.logger.error(
-                    'Cannot create the corresponding add_role request for %s: %s' % (self.request.reqid, e)
-                )
-
-            self.review_messages['declined'] = declined_msg
+            self.__ensure_add_role_request(source_project)
```

Comment:
Not exactly what I meant. :wink:

Now the `ensure_add_role_request` function does two different things, the one suggested by its name and also declining the request. Declining the request inside a function call is inconsistent with the rest of `check_source_submission` (in which that is done explicitly in the body of the main function).

I was thinking on something like:

```python
if not self.source_has_correct_maintainers(source_project):
    declined_msg = (
        'This request cannot be accepted unless %s is a maintainer of %s.' %
        (self.required_maintainer, source_project)
    )

    add_role_req = self._ensure_add_role_request(source_project)
    if add_role_req:
        declined_msg += ' Created the add_role request %s for addressing this problem.' % add_role_req.reqid
    self.review_messages['declined'] = declined_msg
```

Reply:
well, I try to shorten up more, but if this is more readable for you, I can adapt it.